### PR TITLE
#203 Added explanation to empty dashboard

### LINKED ISSUE #203: Add explanation/instructions to empty dashboard
See https://github.com/OpenLiberty/liberty-tools-eclipse/issues/294

### PR DESCRIPTION
**Pull request by Rahul-Biju**
Fixes #203 Added explanation to empty dashboard

### CODE REVIEW

File: src/liberty/libertyProject.ts [hunk 1]
```diff
@@ -213,6 +213,10 @@ export class ProjectProvider implements vscode.TreeDataProvider<LibertyProject>
 
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars
 	public async getChildren(element?: LibertyProject): Promise<LibertyProject[]> {
+		
+		if (this.projects.size <= 0) {
+			return Promise.resolve([]);
+		}
```
Comment (TrevCraw) [inline src/liberty/libertyProject.ts:219]: what is this code for?

File: package.nls.json [hunk 2]
```diff
@@ -12,5 +12,6 @@
     "commands.title.view.test.report": "View test report",
     "commands.title.add.project": "Add project to Liberty Dashboard",
     "commands.title.remove.project": "Remove project from Liberty Dashboard",
-    "commands.title.show.commands": "Show Liberty commands"
+    "commands.title.show.commands": "Show Liberty commands",
+    "views.welcome.message":" \n 1. If no projects are open in the VS Code Explorer, open or create a Liberty project using the File menu. \n 2. If there is one or more existing Maven or Gradle projects open in the VS Code Explorer, try one of the following: \n -Configure the Liberty build plugin in the build file of an existing Maven or Gradle project.\n - Add a server.xml file to an existing Maven or Gradle project at “src/main/liberty/config”.\n - Manually add an existing Maven or Gradle project to the Liberty Dashboard through the VS Code command palette action “Liberty: Add project to Liberty Dashboard”. "
```
Comment (TrevCraw) [inline package.nls.json:16]: - If no projects are open in the VS Code Explorer, **open** or **create** a Liberty project using the File menu. (Can we add an action link for "open" and "create"?)
  - If there is one or more existing Maven or Gradle projects open in the VS Code Explorer, try one of the following:
    - Configure the Liberty build plugin in the build file of an existing Maven or Gradle project. [Click here for more Maven info](https://github.com/OpenLiberty/ci.maven/#configuration) or [click here for more Gradle info](https://github.com/OpenLiberty/ci.gradle#adding-the-plugin-to-the-build-script).
    - **Add a server.xml** file to an existing Maven or Gradle project at “src/main/liberty/config”. [Click here for more info](https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html#server-xml). (Can we add an action link for “Add a server.xml file” where we generate an empty server.xml for the user?)
    - Manually add an existing Maven or Gradle project to the Liberty Dashboard through the VS Code command palette action “**Liberty: Add project to Liberty Dashboard**”. (Can we add an action link for “Liberty: Add project to Liberty Dashboard”?)
cc @aparnamichael

**Review by TrevCraw — CHANGES_REQUESTED** [review]

File: src/liberty/libertyProject.ts (hunk 1 above)
Comment (Rahul-Biju) [reply src/liberty/libertyProject.ts:219]: initially i thought we need to add this check in getChildren() in order to get the welcome view. But now i have tested and found that this check is not necessary for the welcome view. Whenever tree view is empty, vscode itself is showing the welcome view. So removing this condition.

File: package.nls.json (hunk 2 above)
Comment (Rahul-Biju) [reply package.nls.json:16]: (Can we add an action link for "open" and "create"?)
**I have added open project option. What is the "create" option , how should it work?**

(Can we add an action link for “Add a server.xml file” where we generate an empty server.xml for the user?)
**There is no build in command for create a file from welcome view. So i was not able to add this option.**

**All the remaining changes are done.**

Screenshot 2023-11-24 at 7 41 05 PM

**Is this fine?**

Comment (TrevCraw) [reply package.nls.json:16]: The screen shot looks OK, but can you please push your local changes to the branch on your fork so that I can try them out?

Comment (Rahul-Biju) [reply package.nls.json]: @TrevCraw 
Push :**Don**e

**Review by aparnamichael — APPROVED** [review]
Changes looks fine to me

File: package.nls.json [hunk 3]
```diff
@@ -12,5 +12,7 @@
...
-    "commands.title.show.commands": "Show Liberty commands"
+    "commands.title.show.commands": "Show Liberty commands",
+    "views.welcome.message":"\n 1. If no projects are open in the VS Code Explorer, open or create a Liberty project using the File menu.\n[Open Project](command:vscode.openFolder) \n 2. If there is one or more existing Maven or Gradle projects open in the VS Code Explorer, try one of the following:\n a. Configure the Liberty build plugin in the build file of an existing Maven or Gradle project. [Click here for more Maven info](https://github.com/OpenLiberty/ci.maven/#configuration) or [Click here for more Gradle info.](https://github.com/OpenLiberty/ci.gradle#adding-the-plugin-to-the-build-script)\n b. Add a server.xml file to an existing Maven or Gradle project at “src/main/liberty/config”. [Click here for more info.](https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html#server-xml)\n c. Manually add an existing Maven or Gradle project to the Liberty Dashboard through the VS Code command palette action “Liberty: Add project to Liberty Dashboard”.\n [Add project to Liberty Dashboard](command:liberty.dev.add.project)"
```
Comment (dmuelle) [inline package.nls.json:16]:
```suggestion
    "views.welcome.message":"\n 1. If no projects are open in the VS Code Explorer, open or create a Liberty project using the File menu.\n[Open Project](command:vscode.openFolder) \n 2. If one or more existing Maven or Gradle projects are open in the VS Code Explorer, try one of the following:\n a. Configure the Liberty build plugin in the build file of an existing Maven or Gradle project. [Click here for more Maven information](https://github.com/OpenLiberty/ci.maven/#configuration) or [Click here for more Gradle information.](https://github.com/OpenLiberty/ci.gradle#adding-the-plugin-to-the-build-script)\n b. Add a server.xml file to an existing Maven or Gradle project at “src/main/liberty/config”. [Click here for more information.](https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html#server-xml)\n c. Manually add an existing Maven or Gradle project to the Liberty Dashboard through the VS Code command palette action “Liberty: Add project to Liberty Dashboard”.\n [Add project to Liberty Dashboard](command:liberty.dev.add.project)"
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review

File: package.nls.json (hunk 3 above)
Comment (Rahul-Biju) [reply package.nls.json]: Changes updated.

File: package.nls.json [hunk 4]
```diff
@@ -12,5 +12,7 @@
...
+    "commands.title.show.commands": "Show Liberty commands",
+    "views.welcome.message": "\n 1. If no projects are open in the VS Code Explorer, open or create a Liberty project using the File menu.\n[Open Project](command:vscode.openFolder) \n 2. If one or more existing Maven or Gradle projects are open in the VS Code Explorer, try one of the following:\n a. Configure the Liberty build plugin in the build file of an existing Maven or Gradle project. [Click here for more Maven information](https://github.com/OpenLiberty/ci.maven/#configuration) or [Click here for more Gradle information.](https://github.com/OpenLiberty/ci.gradle#adding-the-plugin-to-the-build-script)\n b. Add a server.xml file to an existing Maven or Gradle project at “src/main/liberty/config”. [Click here for more information.](https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html#server-xml)\n c. Manually add an existing Maven or Gradle project to the Liberty Dashboard through the VS Code command palette action “Liberty: Add project to Liberty Dashboard”.\n [Add project to Liberty Dashboard](command:liberty.dev.add.project)"
```
Comment (dmuelle) [inline package.nls.json:16]:
```suggestion
    "views.welcome.message": "\n 1. If no projects are open in the VS Code Explorer, open or create a Liberty project using the File menu.\n[Open Project](command:vscode.openFolder) \n 2. If one or more existing Maven or Gradle projects are open in the VS Code Explorer, try one of the following actions:\n a. Configure the Liberty build plugin in the build file of an existing [Maven](https://github.com/OpenLiberty/ci.maven/#configuration) or [Gradle](https://github.com/OpenLiberty/ci.gradle#adding-the-plugin-to-the-build-script) project.\n b. Add a [server.xml file](https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html#server-xml) to an existing Maven or Gradle project at “src/main/liberty/config”. \n c. Manually add an existing Maven or Gradle project to the Liberty Dashboard through the VS Code command palette action “Liberty: Add project to Liberty Dashboard”.\n [Add project to Liberty Dashboard](command:liberty.dev.add.project)"
```

**Review by dmuelle — APPROVED** [review]
LGTM

**Review by yeekangc — APPROVED** [review]

**Pull request merged**